datalad/support/archives.py

```diff
@@ -259,7 +259,8 @@ def __init__(self, archive, path=None, persistent=False):
         self._archive = archive
         # TODO: bad location for extracted archive -- use tempfile
         if not path:
-            path = tempfile.mktemp(**get_tempfile_kwargs(prefix=_get_cached_filename(archive)))
+            fd, path = tempfile.mkstemp(**get_tempfile_kwargs(prefix=_get_cached_filename(archive)))
+            os.close(fd)
```

Comment:
more fails remain e.g.

```
FAILED ../datalad/tests/test_archives.py::test_ExtractedArchive - RuntimeError: Directory /home/appveyor/DLTMP/datalad_temp_e319196a26_htmijjms already exists whenever it should not persist
```
so there seems to be some side effect remaining... it seems that we need to fixup code...  seems can't do suggestion on lines not modified by PR but I think it should be sufficient to bring `if` below to be `elif` since it makes sense only for cases when path is not specified.

Reply:
I ran tests locally on the test_core.py - found 32 failed, 6 success, and 3 skipped on the base version. So, I doubt the rest of the failures are due to the first patch.

Still, I'll look into it and share concerns.

Reply:
1. I'm wondering if `self.path` and `self._path` are meant to be the same thing in `datalad/support/archives.py`. Because I'm not seeing any code that stores value to `self.path`. If it's a typo, then we can refactor the codebase to work with proposed `mkstemp()` method.

2. The `RunTimeError` is taking place due to the implementation at [line 265](https://github.com/datalad/datalad/blob/maint/datalad/support/archives.py#L265). It's not related to tests or anything.

Reply:
pushed a "fix" so we could finalize this PR